src/FileIndexer.ts

```diff
@@ -104,7 +108,10 @@ export class FileIndexer {
     if (isDefinitionNode) {
       role |= scip.scip.SymbolRole.Definition
     }
-    for (const declaration of sym?.declarations || []) {
+    const declarations = isDefinitionNode
+      ? [node.parent] // Don't emit ambiguous definition at definition-site
+      : sym?.declarations || []
```

Comment:
This comment doesn't properly explain what is going on. Why does `sym?.declarations` lead to ambiguity? Can we trust it for non-definition nodes? It seems like there is something unusual going on with the AST structure here which needs more explanation.

Reply:
You can reproduce the ambiguous definition with tsserver

```
export const Conflict = 42
interface Conflict {}
          ^^^^^^^^ Go to definition here shows two results: the const and the interface
```

I updated the comment to give this example and link to this PR. 

> Can we trust it for non-definition nodes?

`sym.declarations` is what tsserver uses so I think we need to trust it except for odd-cases like this one where it seems like the API has a bug.